### ISSUE TITLE
Linux AL Language extension support

### ISSUE THREAD

**Issue by stekatt** [microsoft/AL#5937]
**Describe the bug**
I have installed vscode on my machine (running linux) but I cannot get the AL Language Extension to work properly. I can only find AL: Open Events Recorder, AL: Go! and AL: Open Externally in the ctrl-shift-P menu. I lack functions to publish my extensions, download symbols and etc.

So my question is if anyone knows if Linux functionallity is on the agenda for the AL Language extension? 

**To Reproduce**

**Expected behavior**

**Screenshots**

**5. Versions:**

 - AL Language: 5.0.280447

 - Business Central: Any

**Comment by dtkb** [comment]
Isn't that normal, until you have an AL project open?

**Comment by kalberes** [comment]
We do not have official support for Linux. But we have tested it internally and it has worked. 
As mentioned you need to have an active project, by opening an AL file that the language server is spawned.

**Comment by atoader** [comment]
One more thing worth mentioning, is that, although we have tested it, we are not shipping the build that will work on Linux. Please go to BCIdeas and create an idea to add support for Linux or vote on one if it already exists.

**Comment by stekatt** [comment]
Thank you both for your answers!
Unfortunately having an open project doesn't seem to help for me. Running al.go results in error "Command 'AL: Go!' resulted in an error (command 'al.go' not found)", just like it was for this person back in 2017 running the extension before support for MacOS was released. See https://github.com/Microsoft/AL/issues/366

@kalberes 
Please let me know what setup you had that worked for you.

My best guess is that Linux binaries are not included in the extension, just because what you say that it's not supported at this moment.
Do you think it would be possible to compile the extension for Linux x86-64? I assume that code-wise not much needs to be reworked from the MacOS version. 

5. Versions:
I'm running Linux Mint 19.3 Cinnamon on a old Intel Core i3.
VS Code version is 1.45.1.
AL Language: 5.0.280447

![screenshot vscode2](https://user-images.githubusercontent.com/66324210/83942043-927ee200-a7f0-11ea-8a2b-6493e88cd2e9.png)

**Comment by dpoe2016** [comment]
Very appreciated to get support for Linux! Thank you!

**Comment by dallyger** [comment]
I've come across this ticket multiple times now, so I searched the "BCIdeas" Ticket and this one seems to be the related one:
<https://experience.dynamics.com/ideas/idea/?ideaid=980a7e1c-10b9-ea11-bf21-0003ff68f7d5>
(Took me a while to find it)
There it suggested that it should work on Linux (see pinned message there), and indeed, it works! Thank you! This was not the case last time I've checked.

If anyone else struggles to get this working (like me first), you must switch to the "Pre-Release" version of the plugin in VS Code. This can be done on the extension's view. There should be a button called "Switch to Pre-Release Version".

I was running the "AL: Go!" command (as suggested here: <https://learn.microsoft.com/en-us/dynamics365/business-central/dev-itpro/developer/devenv-get-started>) using the stable version, and the following error message occurred:

```
command 'al.go' not found
```

Switching to the pre-release one fixed it.
